### DISCUSSION TITLE
Section Categories

### DISCUSSION THREAD

**Discussion by Chwbaccaa** [AdiAddons/AdiBags#898] [Q&A] [answered]
Hey all,
LOVE this addon! Thanks for all the hard work.

I've a quick question, is there a way for us to add or change the names of the Section Categories?

I'm a BIG crafter in WoW and I'm running into issues with the addon slowing down as I add new tiems to a category.

Ideally, I'd like to set up my categories something like...

(Section Category) Tailoring
 -DF Tailoring
 -SL Tailoring
 -WOTLK - Tailoring
 ...etc
(Section Category) Black Smithing
 -DF Black Smithing
 -SL Black Smithing
 -WOTLK - Black Smithing
 ...etc
(Section Category) Leatherworking
 -DF Leatherworking
 -SL Leatherworking
 -WOTLK - Leatherworking
 ...etc
 
 The issue I'm having atm, is that there are SO many items under both the Tradeskills & Consumables Categories that it slows my computer down for quite a bit while it processes any changes. I thought perhaps if we could adjust the Section Categories, then the data could be more spread out and not so centralised.

**Comment by Cidan** [comment, marked as answer]
I've actually spent the evening exploring just this, funny enough. Someone in-game asked me for the same feature, and I got curious about how it would be implemented. Unfortunately, it's not quite as simple as I had hoped. I'm working on building a giant index of itemid -> expansion + profession data. There are some addons that implement this partially, but for whatever reason, they index on the expansion and not the item, which makes it unsuitable for the bag addon.

I actually had this working by building an integration with SmexyMats just a few hours ago, but it was very, very slow as SmexyMaps does a loop over every, single, item in it's database, per item lookup. Let me think a bit more on this, but this feedback has been heard loud and clear.

Thanks :)

**Comment by Cidan** [comment]
Alright, I have an extremely fast method of doing this right now, but I'm borrowing someone else's data to do this. Below you'll see how I have things sorted into Dragonflight (but no sub category yet). This should be relatively easy to do, but the tough part here is getting the data in place. More soon :)

![image](https://user-images.githubusercontent.com/1133628/211474880-898aa7fe-d814-4828-9ec3-3bdc00d85361.png)

**Reply by Chwbaccaa** [reply]
Awesome! I have a lot of the older stuff organised somewhat. If there's a way for me to export the list, I could send it to U. Let me know.

**Reply by Chwbaccaa** [reply]
OH Also, I think "Sorted" the addon has a massive list if U can rip that. I'm not certain how correct it is.

**Comment by Cidan** [comment]
So, it turns out that for Dragonflight, Blizzard added an API that gives you a lot more information on items, specifically on reagent data! Check it out:

![image](https://user-images.githubusercontent.com/1133628/211486530-ffa9fd92-7858-4e08-88d3-d0e0ed57115a.png)

I did all of this using built in Blizzard API's, and it's fast! I need to make a few changes/options to the filter plugin, but I'll likely push this sometime this week :)

**Reply by Chwbaccaa** [reply]
OUTSTANDING!! Thanks so much!

**Comment by Cidan** [comment]
![image](https://user-images.githubusercontent.com/1133628/211492974-429d9ba9-606f-4000-863e-f309c42f1d4f.png)

One more image -- reagent bank looking much, much better!

**Reply by Chwbaccaa** [reply]
Awesome!!
